lib/rucio/core/did.py

```diff
@@ -692,7 +694,19 @@ def delete_dids(dids, account, expire_rules=False, session=None, logger=logging.
         with record_timer_block('undertaker.parent_content'):
             for parent_did in session.query(models.DataIdentifierAssociation).filter(or_(*parent_content_clause)):
                 existing_parent_dids = True
-                detach_dids(scope=parent_did.scope, name=parent_did.name, dids=[{'scope': parent_did.child_scope, 'name': parent_did.child_name}], session=session)
+                detach_dids(scope=parent_did.scope, name=parent_did.name, dids=[{'scope': parent_did.child_scope, 'name': parent_did.child_type}], session=session)
+
+    # Set Epoch tombstone for the files replicas inside the did
+    if config_core.get('undertaker', 'remove_all_files', default=False, session=session) and file_content_clause:
+        with record_timer_block('undertaker.file_content'):
+            file_replicas_clause = [and_(models.RSEFileAssociation.scope == cont['child_scope'], models.RSEFileAssociation.name == cont['child_name']) for cont in session.query(models.DataIdentifierAssociation).filter(or_(*file_content_clause))]
+            none_value = None  # Hack to get pep8 happy
+            for chunk in chunks(file_replicas_clause, 100):
```

Comment:
Doesn't this ignore the case where the file is also in another dataset? In that case the file is a correct secondary replica?